src/pyodide/python-entrypoint-helper.js

```diff
@@ -210,16 +210,26 @@ async function setupPackages(pyodide) {
   return pyodide.pyimport(mainModule);
 }
 
+let mainModule;
+async function getMainModule() {
+  if (mainModule) {
```

Comment:
```suggestion
  if (mainModule === undefined) {
```

Reply:
If this method ends up getting called multiple times without `await` we'll end up with multiple calls to `loadPyodide` running. If the call is already inflight, in would be good to just attach to the existing promise.

Reply:
How's it look now?